contracts/Token.sol

```diff
@@ -111,54 +111,50 @@ contract ShieldToken is OwnableUpgradeable, ERC20PausableUpgradeable {
             totalAmount,
             monthlyAmount,
             initialAmount,
-            releaseTime + afterDays,
-            afterDays,
-            true,
-            monthDelay
+            lockDaysPeriod,
+            true
         );
 
         // Add wallet to frozen wallets
         frozenWallets[wallet] = frozenWallet;
     }
 
-    // function getTimestamp() external view returns (uint256) {
-    //     return block.timestamp;
-    // }
-
-    function getMonths(uint afterDays, uint monthDelay) public view returns (uint) {
-        uint time = releaseTime + afterDays;
+    // this function returns upper rounded amount of months.
+    // 0 - locked
+    // 1 - unlock initial amount (vesting not available yet)
+    // x... - months since lockup period is over (vesting months == x - 1)
+    function getMonths(uint256 lockDaysPeriod) public view returns (uint) {
+        uint unlockTime = releaseTime + lockDaysPeriod;
 
-        if (block.timestamp < time) {
+        if (block.timestamp < unlockTime) {
             return 0;
         }
 
-        uint diff = block.timestamp - time;
-        uint months = diff / 30 days + 1 - monthDelay;
+        uint diff = block.timestamp - unlockTime;
+        uint months = diff / 30 days + 1;
 
         return months;
     }
 
-    function isStarted(uint startDay) public view returns (bool) {
-        if (block.timestamp < releaseTime || block.timestamp < startDay) {
-            return false;
-        }
+    function getTransferableAmount(address sender) public view returns (uint256) {
+        uint256 months = getMonths(frozenWallets[sender].lockDaysPeriod);
 
-        return true;
-    }
+        // lockup period
+        if (months == 0) {
+            return 0;
+        }
 
-    function getTransferableAmount(address sender) public view returns (uint256) {
-        uint months = getMonths(frozenWallets[sender].afterDays, frozenWallets[sender].monthDelay);
-        uint256 monthlyTransferableAmount = frozenWallets[sender].monthlyAmount * months;
-        uint256 transferableAmount = monthlyTransferableAmount + frozenWallets[sender].initialAmount;
+        // TODO: overflow if months becomes too high (wontfix because it would take too long?)
```

Comment:
to make an overflow with current params we have to wait longer than lifespan Sun 😉 

however,  we can add extra check in case if we want to use it with different params.